minimint-bridge/src/client.rs

```diff
@@ -50,6 +50,17 @@ impl Client {
             .insert_entry(&PaymentKey(payment.invoice.payment_hash().clone()), payment)
             .expect("Db error");
     }
+
+    pub fn update_payment_status(&self, payment_hash: &sha256::Hash, status: PaymentStatus) {
+        if let Some(mut payment) = self.fetch_payment(&payment_hash) {
+            payment.status = status;
+            self.client
+                .db()
+                .insert_entry(&PaymentKey(*payment_hash), &payment)
+                .expect("Db error");
+        }
+        // TODO: what to do if this payment doesn't exist?
+    }
```

Comment:
This fixes that problem where timestamps were getting overwritten.

Now the poller will just look up existing payment entry in the database and only overwrite the status.

Reply:
Maybe `PaymentStatus::Paid` should contain a timestamp inside it? That timestamp would be the time it was paid? At least the time we observed that it was paid. Ideally the backend would tell us this timestamp ...